tests/Crypto/tst_cryptorequests/tst_cryptorequests.cpp

```diff
@@ -280,125 +264,131 @@ void tst_cryptorequests::getPluginInfo()
+        QCOMPARE(randomData.size(), generateRandomDataSize);
+        QVERIFY(!allCharactersAreNull(randomData));
+    }
+
+    if (pluginSupportsSeedRandomDataGeneratorRequest) {
+        // test seeding the random number generator
+        SeedRandomDataGeneratorRequest srdgr;
+        srdgr.setManager(&m_cm);
+        QSignalSpy srdgrss(&srdgr, &SeedRandomDataGeneratorRequest::statusChanged);
+        srdgr.setCryptoPluginName(plugins.value(CryptoTest::CryptoPlugin));
+        QCOMPARE(srdgr.cryptoPluginName(), plugins.value(CryptoTest::CryptoPlugin));
+        srdgr.setCsprngEngineName(csprngEngineName);
+        QCOMPARE(srdgr.csprngEngineName(), csprngEngineName);
+        srdgr.setSeedData(seedData);
+        QCOMPARE(srdgr.seedData(), seedData);
+        srdgr.setEntropyEstimate(seedRandomDataEntropyEstimate);
+        QCOMPARE(srdgr.entropyEstimate(), seedRandomDataEntropyEstimate);
+        QCOMPARE(srdgr.status(), Request::Inactive);
+        srdgr.startRequest();
+        QCOMPARE(srdgrss.count(), 1);
+        WAIT_FOR_REQUEST_SUCCEEDED(srdgr);
+        QCOMPARE(srdgrss.count(), 2);
+
+        if (pluginSupportsGenerateRandomDataRequest) {
+            // ensure that we get different random data to the original set
+            grdr.startRequest();
+            WAIT_FOR_REQUEST_SUCCEEDED(grdr);
+            QByteArray seededData = grdr.generatedData();
+            QCOMPARE(seededData.size(), generateRandomDataSize);
+            QVERIFY(randomData != seededData);
         }
     }
-    QVERIFY(!allNull);
-
-    // test seeding the random number generator
-    SeedRandomDataGeneratorRequest srdgr;
-    srdgr.setManager(&cm);
-    QSignalSpy srdgrss(&srdgr, &SeedRandomDataGeneratorRequest::statusChanged);
-    srdgr.setCryptoPluginName(DEFAULT_TEST_CRYPTO_PLUGIN_NAME);
-    QCOMPARE(srdgr.cryptoPluginName(), DEFAULT_TEST_CRYPTO_PLUGIN_NAME);
-    srdgr.setCsprngEngineName(GenerateRandomDataRequest::DefaultCsprngEngineName);
-    QCOMPARE(srdgr.csprngEngineName(), GenerateRandomDataRequest::DefaultCsprngEngineName);
-    srdgr.setSeedData(QByteArray("seed"));
-    QCOMPARE(srdgr.seedData(), QByteArray("seed"));
-    srdgr.setEntropyEstimate(0.5);
-    QCOMPARE(srdgr.entropyEstimate(), 0.5);
-    QCOMPARE(srdgr.status(), Request::Inactive);
-    srdgr.startRequest();
-    QCOMPARE(srdgrss.count(), 1);
-    QCOMPARE(srdgr.status(), Request::Active);
-    QCOMPARE(srdgr.result().code(), Result::Pending);
-    WAIT_FOR_FINISHED_WITHOUT_BLOCKING(srdgr);
-    QCOMPARE(srdgrss.count(), 2);
-    QCOMPARE(srdgr.status(), Request::Finished);
-    QCOMPARE(srdgr.result().code(), Result::Succeeded);
-
-    // ensure that we get different random data to the original set
-    grdr.startRequest();
-    WAIT_FOR_FINISHED_WITHOUT_BLOCKING(grdr);
-    QByteArray seededData = grdr.generatedData();
-    QCOMPARE(seededData.size(), 2048);
-    QVERIFY(randomData != seededData);
+}
 
+void tst_cryptorequests::randomData_dev_urandom()
```

Comment:
This method can probably be removed, and should be replaced by simply adding a new row to the randomData_data() above which sets the engine to "/dev/urandom", I think

Reply:
We could, though this test checks the generated random number, which isn't done in the randomData() test since that just accepts any non-null byte array, so we would lose that in the test.